src/oop/special_class_method/hard/q8/MainTest.java

```diff
@@ -0,0 +1,325 @@
+package oop.special_class_method.hard.q8;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest {
+    // Java
+    private final String className = "Circle";
+    private final String doubleFieldName = "radius";
+    private final String stringFieldName = "colour";
+    private final String setStringMethodName = "setColour";
+    private final String getStringMethodName = "getColour";
+    private final String getDoubleMethodName = "getRadius";
+    private final String setDoubleMethodName = "setRadius";
+    private final String testClassName = "TestClass";
+    private final double outputRadius = 3.75;
+    private final String outputColour = "Red";
+    private ObjectTest testClass;
+    private ObjectTest outputClass;
+
+    @BeforeEach
+    public void setUp() {
+        String classString = "oop.special_class_method.hard.q8." + className;
+        testClass = new ObjectTest(classString);
+        String outputClassString = "oop.special_class_method.hard.q8." + testClassName;
+        outputClass = new ObjectTest(outputClassString);
+    }
+
+    @Test
+    @Tag("dependent1")
+    @Tag("dependent2")
+    public void correctTestClass() throws Throwable {
```

Comment:
The main method here should be **dependency1**, and all methods it is dependent on should have the tag **dependent1**